src/main/java/cn/nukkit/entity/EntityHuman.java

```diff
@@ -50,14 +52,21 @@ public float getHeight() {
         return 1.8f;
     }
 
+    @Since("FUTURE")
+    @PowerNukkitOnly
+    @Override
+    public float getSwimmingHeight() {
+        return getWidth();
+    }
+
     @Override
     public float getEyeHeight() {
-        return 1.62f;
+        return (float)(boundingBox.getMaxY() - boundingBox.getMinY() - 0.18);
     }
 
     @Override
     protected float getBaseOffset() {
-        return this.getEyeHeight();
+        return 1.62f;
     }
 
```

Comment:
On this change i'm confused. why did getBaseOffset got inlined to `1.62f` when previously it was the same as `getEyeHeight`, while getEyeHeight became a complex calculation.

From this i can spot a problem that could arise.
When a player is swimming and a plugin requests `getBaseOffset` the offset given to it could desync with the actual player height (since its a set inline value, and the bouding box now changes to approx. a meter cubed and 1.62 being +.62 above the aabb). Was this case considered?

Reply:
The `BaseOffset` is the height difference between the packet communication and the actual position on the server. It usually doesn't change and before we could swim, it was the same as the eye height for the human entities, so somebody might have used getEyeHeight as a shortcut.

The `EyeHeight` is where the eyes are located in a height considering that the player is on y 0. When the player starts swimming, the player bounding box reduces in height, so does the eye height. Since this is not a constant value (thank god we have getters here), the value is adjusted depending on the player's current height. I don't know if sneaking also changes the player's height in Bedrock Edition, if it does, then we can also benefit from this change to return the right eye height.

Long story short, both used to return the same  constant values, now only `getBaseOffset()` return the old value and `getEyeHeight()` returns the current player eye height.

Reply:
Understood.